src/main/java/seedu/contax/logic/commands/EditPriorityCommand.java

```diff
@@ -0,0 +1,134 @@
+package seedu.contax.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.contax.commons.core.Messages;
+import seedu.contax.commons.core.index.Index;
+import seedu.contax.logic.commands.exceptions.CommandException;
+import seedu.contax.model.Model;
+import seedu.contax.model.appointment.Appointment;
+import seedu.contax.model.appointment.Priority;
+
+
+/**
+ * Edits the priority level (low, medium or high) of an existing appointment in the appointment list.
+ */
+public class EditPriorityCommand extends Command {
+    public static final String COMMAND_WORD = "priority";
+
+    public static final String MESSAGE_USAGE = "`" + COMMAND_WORD
+            + "`: **Edits the level of priority of an existing appointment in the appointment list.**\n"
+            + "Parameters: *INDEX pri/PRIORITY*\n"
+            + "Example: `" + COMMAND_WORD + " 1 " + "pri/high`";
+
+    public static final String MESSAGE_EDIT_PRIORITY_SUCCESS = "Edited Priority of Appointment at index %d: %s";
+    public static final String MESSAGE_REMOVE_PRIORITY_SUCCESS = "Removed Priority of Appointment at index %d";
+    public static final String MESSAGE_PRIORITY_NOT_EDITED =
+            "The priority level you provided is same as existing priority.";
+
+    private final Index index;
+    private final EditPriorityDescriptor editPriorityDescriptor;
+
+    /**
+     * Creates an {@code EditPriorityCommand} object.
+     *
+     * @param index The specified index to update the priority.
+     * @param editPriorityDescriptor Details to edit the priority with.
+     */
+    public EditPriorityCommand(Index index, EditPriorityDescriptor editPriorityDescriptor) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.editPriorityDescriptor = editPriorityDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Appointment> appointmentList = model.getFilteredAppointmentList();
+
+        if (index.getZeroBased() >= appointmentList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPOINTMENT_DISPLAYED_INDEX);
+        }
+
+        Appointment appointment = appointmentList.get(index.getZeroBased());
+
+        if (editPriorityDescriptor.getPriority().isPresent()) {
+            Priority priority = editPriorityDescriptor.getPriority().get();
```

Comment:
Does `Optional.get()` fail with exception if there is no content?

Reply:
shd checked in getPriority().isPresent()

Reply:
Sorry, my bad on this, I did not see the `isPresent()` check.